src/ie9-ajax.js

```diff
@@ -58,8 +58,17 @@ function ie9Ajax(method, url, opts) {
     xdr.onerror = handleError;
     xdr.ontimeout = handleError;
 
-    xdr.open(method, authToUrl(url, opts));
-    xdr.send(opts.data);
+    try {
+      xdr.open(method, authToUrl(url, opts));
+      xdr.send(opts.data);
+    } catch (error) {
+      // using trim because the error message has a bunch of white space for some reason
+      if (error.message.trim() === 'Access is denied.') {
+        reject(new Error('Please ensure Shopify JavaScript Buy SDK is run over https'));
```

Comment:
I would use the error logger to ensure this error gets properly logged. Unless that's already happening farther up the promise chain.